functions/supplier/debezium-supplier/src/main/java/org/springframework/cloud/fn/supplier/debezium/DebeziumEngineAutoConfiguration.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2023-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.supplier.debezium;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.function.Consumer;
+
+import io.debezium.engine.ChangeEvent;
+import io.debezium.engine.DebeziumEngine;
+import io.debezium.engine.DebeziumEngine.CompletionCallback;
+import io.debezium.engine.DebeziumEngine.ConnectorCallback;
+import io.debezium.engine.spi.OffsetCommitPolicy;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * DebeziumEngine auto-configuration.
+ *
+ * The engine configuration is entirely standalone and only talks with the source system; Applications using the engine
+ * auto-configuration simply provides a {@link Consumer consumer function} implementation to which the engine will pass
+ * all records containing database change events.
+ * <p>
+ * With the engine, the application that runs the connector assumes all responsibility for fault tolerance, scalability,
+ * and durability. Additionally, applications must specify how the engine can store its relational database schema
+ * history and offsets. By default, this information will be stored in memory and will thus be lost upon application
+ * restart.
+ * <p>
+ * Engine Is designed to be submitted to an {@link Executor} or {@link ExecutorService} for execution by a single
+ * thread, and a running connector can be stopped either by calling {@link #stop()} from another thread or by
+ * interrupting the running thread (e.g., as is the case with {@link ExecutorService#shutdownNow()}).
+ *
+ * @author Christian Tzolov
+ */
+@AutoConfiguration
+@EnableConfigurationProperties(DebeziumProperties.class)
+public class DebeziumEngineAutoConfiguration {
+
+	private static final Log logger = LogFactory.getLog(DebeziumEngineAutoConfiguration.class);
+
+	@Bean
+	public Properties debeziumConfiguration(DebeziumProperties properties) {
```

Comment:
Spring Boot dictates a recommendation to not have a `Properties` as a bean.

Consider to have an ctor injection for the `DebeziumProperties` and create an internal instance for this `Properties` to use in bean method below.

Reply:
Replaced the bean with on-the-fly propety method: getDebeziumNativeConfiguration